test_project/sample/models.py

```diff
@@ -116,3 +116,9 @@ class RelatedToItemViaPrimaryKey(models.Model):
 
 class RelatedToItemViaUniqueField(models.Model):
     item = models.ForeignKey(Item, to_field="legacy_id", on_delete=models.CASCADE)
+
+
+class RelatedToItemAssignDb_column(models.Model):
+    item = models.ForeignKey(
+        Item, db_column="item", to_field="legacy_id", on_delete=models.CASCADE
```

Comment:
Why do you specify db_column value if you intent to use the same name as what django would use as default ?
```suggestion
        Item, db_column="item", to_field="legacy_id", on_delete=models.CASCADE
```
This fixes it.
```suggestion
        Item, db_column="item_", to_field="legacy_id", on_delete=models.CASCADE
```
or fixes it also, and the db_column is still "item"
```suggestion
        Item, to_field="legacy_id", on_delete=models.CASCADE
```